pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -703,3 +729,90 @@ func managedPermission(action, resource string, resourceID, resourceAttribute st
 		Scope:  accesscontrol.Scope(resource, resourceAttribute, resourceID),
 	}
 }
+
+/*
+ACTION SETS
+Stores actionsets IN MEMORY
+*/
+// - To grant access to users/teams/basic roles during runtime, users will go through the managed permissions view and access-control will be enforced based on their permissions.
+// ActionSet is a struct that represents a set of actions that can be performed on a resource.
+// An example of an action set is "folders:edit" which represents the actions that can be performed on a datasource with the uid of 1.
+/*
+```go
+actionSet := &ActionSet{
+	Resource:   "folders",
+	Permission: "edit",
+	Actions:    []string{"folders:read", "folders:write", "dashboards:read", "dashboards:write"},
+}`
+*/
+type ActionSet struct {
+	Action     string   `json:"action"`
+	Resource   string   `json:"resource"`
+	Permission string   `json:"permission"`
+	Actions    []string `json:"actions"`
+}
+
+// InMemoryActionSets is an in-memory implementation of the ActionSetService.
+type InMemoryActionSets struct {
+	log        log.Logger
+	actionSets map[string]*ActionSet
+}
+
+// NewInMemoryActionSets returns a new instance of InMemoryActionSetService.
+func NewInMemoryActionSets(log log.Logger) *InMemoryActionSets {
+	return &InMemoryActionSets{
+		actionSets: make(map[string]*ActionSet),
+		log:        log,
+	}
+}
+
+// GetActionSet returns the action set for the given action.
+func (s *InMemoryActionSets) GetActionSet(actionName string) *ActionSet {
+	actionSet, ok := s.actionSets[actionName]
+	if !ok {
+		return nil
+	}
+	return actionSet
+}
+
+func (s *InMemoryActionSets) StoreActionSet(resource, permission string, actions []string) {
+	// whitelist the resources and permissions
+	if (resource != "folders") || (permission != "admin" && permission != "edit" && permission != "view") {
+		s.log.Debug("Not able to action set")
+		return
+	}
+	s.log.Debug("storing action set")
+	name, err := s.CreateActionSetName(resource, permission)
+	if err != nil {
+		s.log.Error("error creating action set", "error", err)
+	}
+	// check if the action set already exists
+	if s.GetActionSet(name) != nil {
+		s.log.Debug("action set already exists", name)
+		return
+	}
```

Comment:
nit: you're storing elements in a map - if the action set exists, it'll simply be overwritten, so I don't think we need this check.

Reply:
is it not interesting that we already have the actionset for now? meaning we have a check for double writing the actionset. I guess that is fine since we have "static" actionsets.